src/interfaces/python/factory_python.i

```diff
@@ -0,0 +1,184 @@
+		{
+			return on_impl(v, SHOGUN_GET_SWIG_TYPE(NeuralLayer));
+		}
+		virtual void on(std::shared_ptr<SplittingStrategy>* v)
+		{
+			return on_impl(v, SHOGUN_GET_SWIG_TYPE(SplittingStrategy));
+		}
+		virtual void on(std::shared_ptr<Pipeline>* v)
+		{
+			return on_impl(v, SHOGUN_GET_SWIG_TYPE(Pipeline));
+		}
+		virtual void on(std::shared_ptr<SVM>* v)
+		{
+			return on_impl(v, SHOGUN_GET_SWIG_TYPE(SVM));
+		}
+		virtual void on(std::shared_ptr<LikelihoodModel>* v)
+		{
+			return on_impl(v, SHOGUN_GET_SWIG_TYPE(LikelihoodModel));
+		}
+		virtual void on(std::shared_ptr<MeanFunction>* v)
+		{
+			return on_impl(v, SHOGUN_GET_SWIG_TYPE(MeanFunction));
+		}
+		virtual void on(std::shared_ptr<DifferentiableFunction>* v)
+		{
+			return on_impl(v, SHOGUN_GET_SWIG_TYPE(DifferentiableFunction));
+		}
+		virtual void on(std::shared_ptr<Inference>* v)
+		{
+			return on_impl(v, SHOGUN_GET_SWIG_TYPE(Inference));
+		}
+		virtual void on(std::shared_ptr<LossFunction>* v)
+		{
+			return on_impl(v, SHOGUN_GET_SWIG_TYPE(LossFunction));
+		}
+		virtual void on(std::shared_ptr<Tokenizer>* v)
+		{
+			return on_impl(v, SHOGUN_GET_SWIG_TYPE(Tokenizer));
+		}
+		virtual void on(std::shared_ptr<CombinationRule>* v)
+		{
+			return on_impl(v, SHOGUN_GET_SWIG_TYPE(CombinationRule));
+		}
+		virtual void on(std::shared_ptr<KernelNormalizer>* v)
+		{
+			return on_impl(v, SHOGUN_GET_SWIG_TYPE(KernelNormalizer));
+		}
+		virtual void on(std::shared_ptr<Transformer>* v)
+		{
+			return on_impl(v, SHOGUN_GET_SWIG_TYPE(Transformer));
+		}
+		virtual void on(std::shared_ptr<MachineEvaluation>* v)
+		{
+			return on_impl(v, SHOGUN_GET_SWIG_TYPE(MachineEvaluation));
+		}
+		virtual void on(std::shared_ptr<StructuredModel>* v)
+		{
+			return on_impl(v, SHOGUN_GET_SWIG_TYPE(StructuredModel));
+		}
+		virtual void on(std::shared_ptr<FactorType>* v)
+		{
+			return on_impl(v, SHOGUN_GET_SWIG_TYPE(FactorType));
+		}
+		virtual void on(std::shared_ptr<ParameterObserver>* v)
+		{
+			return on_impl(v, SHOGUN_GET_SWIG_TYPE(ParameterObserver));
+		}
+		virtual void on(std::shared_ptr<Distribution>* v)
+		{
+			return on_impl(v, SHOGUN_GET_SWIG_TYPE(Distribution));
+		}
+		virtual void on(std::shared_ptr<GaussianProcess>* v)
+		{
+			return on_impl(v, SHOGUN_GET_SWIG_TYPE(GaussianProcess));
+		}
+		virtual void on(std::shared_ptr<Alphabet>* v)
+		{
+			return on_impl(v, SHOGUN_GET_SWIG_TYPE(Alphabet));
+		}
+
+		template <typename T>
+		void on_impl(std::shared_ptr<T>* v, swig_type_info* type)
+		{
+			if (!v)
+				return;
+
+			auto smartresult = new std::shared_ptr<T>(*v);
+			m_pyobj = SWIG_Python_NewPointerObj(
+				nullptr, SWIG_as_voidptr(smartresult), type, SWIG_POINTER_OWN);
+		}
+
+		PyObject* pyobj()
+		{
+			// this should never be true since "create" throws on error
+			if (!m_pyobj)
+			{
+				m_pyobj = SWIG_Python_NewPointerObj(
+					nullptr, nullptr, SHOGUN_GET_SWIG_TYPE(SGObject),
+					SWIG_POINTER_OWN);
+			}
+			return m_pyobj;
```

Comment:
even if it *should* not be possible to reach this, could you replace this with an exception? The worst thing that can happen is segfault in Python, because then the interpreter crashes. An error can at least be handled gracefully

Reply:
ah nvm, I see that you handle that below

Reply:
In this case I return a python null object. I will change the function call to `SWIG_Py_Void()` to be more clear and add a comment.

Reply:
Scratch that. Throwing is better.